dropwizard-logging/src/main/java/smartthings/dw/logging/turbo/SamplingTurboFilter.java

```diff
@@ -0,0 +1,50 @@
+package smartthings.dw.logging.turbo;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.turbo.TurboFilter;
+import ch.qos.logback.core.spi.FilterReply;
+import org.slf4j.Marker;
+import org.slf4j.MarkerFactory;
+
+import java.util.Random;
+
+public class SamplingTurboFilter extends TurboFilter {
+
+    public final static Marker SAMPLING_MARKER = MarkerFactory.getMarker("sampling");
+
+    private final static int SAMPLE_RANGE = 10000;
+    private final static double MINIMAL_SAMPLE_RATE = 1.0 / SAMPLE_RANGE;
+
+    private long boundary = SAMPLE_RANGE;
+
+    private Random rand = new Random();
```

Comment:
@edwardsoo-ss can we switch this to use ThreadLocalRandom instead?  Assuming this will run on multiple threads correct? Not entirely sure how the filters in slf4j/logback work I guess.